dist/docker/Dockerfile.ubuntu

```diff
@@ -52,19 +52,19 @@ COPY . /cloe
 RUN \
     # Export our own Conan recipes, in case they are not available in the
     # CONAN_REMOTE specified above.
-    make export-vendor export && \
+    make export-vendor && \
     # Build all the packages, except for vtd, because that currently requires
     # dependencies we don't have in the Docker container.
     # You can specify more than one package here, see the Makefile for more
     # information on the WITH_*, PACKAGES, NOBUILD_PKGS, and BUILD_PKGS variables.
-    make WITH_VTD=0 package-all && \
+    make package-select && \
```

Comment:
To me it seems that `make package-select` leads to different boost versions being used by different packages. I tried to enforce overriding the boost version from the cloe meta package with the following commands:
```
    make export && \
    make SELECT_PKGS=cloe package-select && \
```
And this is what I get:
```
    boost/1.69.0:8b18088946f796b7dd7fdb66047b6b29a5886f7d - Download
    bzip2/1.0.8:b27e2b40bcb70be2e7d6f0e0c6e43e35c530f8d3 - Download
    cli11/1.9.1:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Download
    cloe/0.18.0-nightly@cloe/develop:fba756a1c0a0755158a7c8c57ca4314d2499a56c - Build
    cloe-engine/0.18.0-nightly@cloe/develop:ae1371a8518682b0ff48942dd3db0d1a2b162c87 - Build
    cloe-models/0.18.0-nightly@cloe/develop:ea39663eebdd8b1fb735b40caf0d381ce7e3ed31 - Build
    cloe-oak/0.18.0-nightly@cloe/develop:06ac97371186bb752613faf501770537907c0f97 - Build
    cloe-plugin-basic/0.18.0-nightly@cloe/develop:47e39a3b7eadc945e378fc3bd964880c6c058314 - Build
    cloe-plugin-gndtruth-extractor/0.18.0-nightly@cloe/develop:5d5cc409208864d014c6a628d496e302b4ba4698 - Build
    cloe-plugin-minimator/0.18.0-nightly@cloe/develop:e303963354aefe834c65497716454ad033873dbd - Build
    cloe-plugin-mocks/0.18.0-nightly@cloe/develop:e303963354aefe834c65497716454ad033873dbd - Build
    cloe-plugin-noisy-sensor/0.18.0-nightly@cloe/develop:47e39a3b7eadc945e378fc3bd964880c6c058314 - Build
    cloe-plugin-speedometer/0.18.0-nightly@cloe/develop:e303963354aefe834c65497716454ad033873dbd - Build
    cloe-plugin-virtue/0.18.0-nightly@cloe/develop:e303963354aefe834c65497716454ad033873dbd - Build
    cloe-runtime/0.18.0-nightly@cloe/develop:a94454f3a44ad77a19548a0b687f9cd8f8355e96 - Build
    cpp-netlib/0.13.0@cloe/stable:e8b64dd6a2bf8eed100549df655533f3a257657e - Build
    eigen/3.3.9:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Download
    fable/0.18.0-nightly@cloe/develop:64066daa151c96dfd5f453dfd4bc950505eb4dd6 - Build
    fmt/6.2.1:66c5327ebdcecae0a01a863939964495fa019a06 - Download
    incbin/1.74.0@cloe/stable:524ea35a8120baabdde02483add58d81bf541327 - Build
    inja/3.0.0:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Download
    libiconv/1.16:d50a0d523d98c15bb147b18fa7d203887c38be8b - Download
    nlohmann_json/3.9.1:d1091b2ed420e6d287293709a907ae824d5de508 - Download
    openssl/1.1.1g:d50a0d523d98c15bb147b18fa7d203887c38be8b - Download
    spdlog/1.5.0:c0fb34b00d4c04a33793ba290326808d0fd5de33 - Download
    zlib/1.2.11:d50a0d523d98c15bb147b18fa7d203887c38be8b - Download
```

Reply:
Not sure that command does what you think it does...

If you want to override boost, you can't use `package-select`, because `package-select` builds each Cloe package separately, so they're all disconnected from `cloe`. In this case it might be best to use:

```
make export package-outdated
```
```

Reply:
My main point was that the docker build fails if you use `make package-select` because of the diamond-type dependency graph of oak (requires cpp-netlib and cloe-runtime, but each of these resolve to a different boost version if you don't override the boost version from the consumer package). 
I can confirm that for both 
`make SELECT_PKGS=cloe package-select` 
and 
`make export package-outdated`
the docker build would work. 
Also, it seems that the default build policy is `outdated` anyway. Shouldn't the two commands result in the same behavior then?

Reply:
Yes, you're right on both points of course.

There are useful use-cases of `package-select`, but the user should be made aware of the caveats.

Reply:
When I understand correctly, we can't use `package-select` then in our docker build, because we need a consistent set of packages, linked to the same version of `boost` which is constrained by our `cloe` metapackage. If the packages are built with `package-select` the metapackage's `boost/[<1.70]` constraint doesn't apply and the build fails at cpp-netlib because that can't be built with boost 1.75 (currently picked by conan from conan-center).

Do we need to decline this improvement or do you have another idea, @cassava?

Reply:
:'-(

Let's decline and re-visit the topic when Conan is able to handle this kind of dependency resolution.